tests/integration_tests/performance/test_versioned_serialization_benchmark.py

```diff
@@ -0,0 +1,147 @@
+
+import platform
+import os
+import logging
+import json
+
+import pytest
+import framework.utils as utils
+import host_tools.proc as proc
+from framework.defs import FC_WORKSPACE_DIR
+
+BENCHMARK_DIRECTORY = "{}/src/vmm".format(FC_WORKSPACE_DIR)
+
+PROC_MODEL = proc.proc_type()
+
+BASELINES = {
+    "Intel": {
+        "serialize": {
+            "no-crc": {
+                "target": 0.146,  # milliseconds
+                "delta": 0.020  # milliseconds
+            },
+            "crc": {
+                "target": 0.186,  # milliseconds
+                "delta": 0.020  # milliseconds
+            }
+        },
+        "deserialize": {
+            "no-crc": {
+                "target": 0.034,  # milliseconds
+                "delta": 0.010  # milliseconds
+            },
+            "crc": {
+                "target": 0.042,  # milliseconds
+                "delta": 0.010  # milliseconds
+            }
+        }
+    },
+    "AMD": {
+        "serialize": {
+            "no-crc": {
+                "target": 0.146,  # milliseconds
+                "delta": 0.020  # milliseconds
+            },
+            "crc": {
+                "target": 0.186,  # milliseconds
+                "delta": 0.020  # milliseconds
+            }
+        },
+        "deserialize": {
+            "no-crc": {
+                "target": 0.034,  # milliseconds
+                "delta": 0.010  # milliseconds
+            },
+            "crc": {
+                "target": 0.042,  # milliseconds
+                "delta": 0.010  # milliseconds
+            }
+        }
+    },
+    "ARM": {
+        "serialize": {
+            "no-crc": {
+                "target": 0.096,  # milliseconds
+                "delta": 0.020  # milliseconds
+            },
+            "crc": {
+                "target": 0.186,  # milliseconds
+                "delta": 0.020  # milliseconds
+            }
+        },
+        "deserialize": {
+            "no-crc": {
+                "target": 0.034,  # milliseconds
+                "delta": 0.010  # milliseconds
+            },
+            "crc": {
+                "target": 0.042,  # milliseconds
+                "delta": 0.010  # milliseconds
+            }
+        }
+    }
+}
+
+
+def _check_statistics(directory, mean):
+    proc_model = [item for item in BASELINES if item in PROC_MODEL]
+    assert len(proc_model) == 1, "Could not get processor model!"
+
+    if "deserialize" in directory.lower():
+        bench = "deserialize"
+    else:
+        bench = "serialize"
+
+    if "crc" in directory.lower():
+        attribute = "crc"
+    else:
+        attribute = "no-crc"
+
+    measure = BASELINES[proc_model[0]][bench][attribute]
+    low = measure["target"] - measure["delta"]
+    high = measure["target"] + measure["delta"]
+    assert low <= mean <= high, "Benchmark result {} has changed!" \
+        .format(directory)
+
+
+@pytest.mark.skipif(
+    platform.machine() != "x86_64",
+    reason="Not supported yet."
```

Comment:
We need to also do this for ARM at some point. @dianpopa can help here.

Reply:
I will contact her.